kessler/kessler_update.F90

```diff
@@ -0,0 +1,118 @@
+module kessler_update
+
+   use ccpp_kinds, only: kind_phys
+
+   implicit none
+   private
+
+   public :: kessler_update_init
+   public :: kessler_update_timestep_init
+   public :: kessler_update_run
+   public :: kessler_update_timestep_final
+
+   ! Private module variables
+   real(kind_phys)    :: gravit
+   real(kind_phys)    :: cpair
+
+CONTAINS
+
+   !> \section arg_table_kessler_update_init  Argument Table
+   !! \htmlinclude kessler_update_init.html
+   subroutine kessler_update_init(gravit_in, cpair_in)
+      real(kind_phys),    intent(in)    :: gravit_in
+      real(kind_phys),    intent(in)    :: cpair_in
+
+      gravit = gravit_in
+      cpair = cpair_in
+
+   end subroutine kessler_update_init
+
+   !> \section arg_table_kessler_update_timestep_init  Argument Table
+   !! \htmlinclude kessler_update_timestep_init.html
+   subroutine kessler_update_timestep_init(temp, temp_prev, ttend_t,          &
+        errmsg, errflg)
+
+      real(kind_phys),    intent(in)  :: temp(:,:)
+      real(kind_phys),    intent(out) :: temp_prev(:,:)
+      real(kind_phys),    intent(out) :: ttend_t(:,:)
+      character(len=512), intent(out) :: errmsg
+      integer,            intent(out) :: errflg
+
+      !   Initialize the previous temperature and its tendency to zero
+      temp_prev(:,:)  = temp(:,:)
+      ttend_t(:,:)    = 0._kind_phys
+
+      errmsg = ''
+      errflg = 0
+
+   end subroutine kessler_update_timestep_init
+
+   !> \section arg_table_kessler_update_run  Argument Table
+   !! \htmlinclude kessler_update_run.html
+   subroutine kessler_update_run(nz, ncol, temp, theta, exner, dt,      &
+        temp_prev, ttend_t, errmsg, errflg)
+
+      integer,            intent(in)    :: nz
+      integer,            intent(in)    :: ncol
+      real(kind_phys),    intent(in)    :: temp(:,:) ! temperature
+      real(kind_phys),    intent(in)    :: exner(:,:)
+      real(kind_phys),    intent(in)    :: dt
+
+      real(kind_phys),    intent(inout) :: theta(:,:)
+      real(kind_phys),    intent(inout) :: temp_prev(:,:)
+      real(kind_phys),    intent(inout) :: ttend_t(:,:)
+
+      character(len=512), intent(out)   :: errmsg
+      integer,            intent(out)   :: errflg
+
+      integer                           :: klev
+      real(kind_phys)                   :: ptend_s(ncol)
+
+      errmsg = ''
+      errflg = 0
+
+      ! Back out tendencies from updated fields
+      do klev = 1, nz
+         ptend_s(:ncol) = ((theta(:ncol,klev) * exner(:ncol,klev)) -          &
+              temp_prev(:ncol,klev)) * cpair / dt
+         ttend_t(:ncol,klev) = ttend_t(:ncol,klev) + (ptend_s(:ncol) / cpair)
+      end do
+
+      ! Save the temperature for the next time step
+      !!XXgoldyXX ==> @cacraigucar: Does this have any effect?
+      temp_prev(:,:)     = temp(:,:)
```

Comment:
It looks like I am just repeating @gold2718's comment here, but is this line needed if we are already setting `temp_prev` in the `kessler_update_timestep_init` subroutine?

Reply:
Removed